### ISSUE TITLE
Function Builder: publish 1.2 with LoL support

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/function-builder#139]
@kathy-phet said that it would be beneficial for Legends of Learning to republish Function Builder from master.  Function Builder is not a PhET-iO sim, so this will be core features.  This also includes Function Builder: Basics.

The plan is:
- [ ] dev test
- [ ] RC test
- [ ] publish 1.1

There are no outstanding issues in the function-builder and function-builder-basics repos, and no general "blocked" issues relevant to these sims. So this work can proceed immediately. 

@amanda-phet FYI.

**Comment by pixelzoom** [comment]
@amanda-phet FYI.

**Comment by pixelzoom** [comment]
The published version is 1.0.x, so I was expecting to be publishing 1.1.x.  But when I went to publish the dev version of Function Builder, it was 1.2.x.  Investigation revealed these 1.1 dev and phet-io (!) versions were published to bayes:

```
drwxrwsr-x+ 3 tomcat     phet  137 Aug 23  2016 1.1.0-dev.1/
drwxrwsr-x+ 3 tomcat     phet  137 Aug 29  2016 1.1.0-dev.2/
drwxrwsr-x+ 2 tomcat     phet  102 May 17  2017 1.1.0-dev.3/
drwxrwsr-x+ 2 tomcat     phet  118 Jun  5  2017 1.1.0-dev.4/
drwxrwxr-x+ 6 tomcat     phet  151 Sep 30  2016 1.1.0-phetio/
drwxrwsr-x+ 5 tomcat     phet  207 Sep 28  2016 1.1.0-phetio.4/
drwxrwxr-x+ 4 tomcat     phet  118 Sep 24  2016 1.1.0-phetiodev.3/
drwxrwxr-x+ 5 tomcat     phet  134 Oct  5  2016 1.1.1-phetio/
drwxrwxr-x+ 5 tomcat     phet  133 Nov 12  2016 1.1.2-phetiophetio/
drwxrwsr-x+ 5 phet-admin phet  205 Jun  5  2018 1.1.3-rc.1/
```

The dates on these are curious.  The initial commit to function-builder was 3/11/17, and all of the phetio versions shown above are dated in 2016.   function-builder also has no PhET-iO implementation, and brand=phet is not enabled in package.json. 

I'm not sure what's going on here, and will investigate a little.  If I can't figure this out, I'm going to advance both function-builder and function-builder-basics  to 1.2, so that their version numbers remain in sync.

**Comment by pixelzoom** [comment]
bayes thinks that it's the correct year, so that's not the problem:

```
% date +%Y
2022
```

**Comment by pixelzoom** [comment]
I still don't know what was going on here, but here's who published these odd versions:

```
Aug 23  2016 1.1.0-dev.1/ => @samreid 
Aug 29  2016 1.1.0-dev.2/ => @samreid
Sep 24  2016 1.1.0-phetiodev.3/ => @samreid 
Sep 28  2016 1.1.0-phetio.4/ => @mattpen
Sep 30  2016 1.1.0-phetio/ => @zepumph
Oct  5  2016 1.1.1-phetio/ => @zepumph 
Nov 12  2016 1.1.2-phetiophetio/ => @samreid 
Jun  5  2018 1.1.3-rc.1/ => @jonathanolson 
May 17  2017 1.1.0-dev.3/ => ?? no commit!
Jun  5  2017 1.1.0-dev.4/ => ?? no commit!
```

Note that the syntax for these version numbers is all over the place.

Do any of you know what was going on here?

**Comment by pixelzoom** [comment]
I still don't know what was going on here, but it looks like there was some PhET-iO work going on that I either didn't know about, or have forgotten about.

@jonathanolson created the 1.1-phetio branch on 07/06/2018.

I see no PhET-iO instrumentation in the 1.1-phetio branch, so I don't know why it was created, or why (or how!) phetio versions were published.

Here's who published these odd versions:

```
drwxrwsr-x+ 2 tomcat     phet 118 Jun  5  2017 1.1.0-dev.4/ => ??
drwxrwsr-x+ 2 tomcat     phet 102 May 17  2017 1.1.0-dev.3/ => ??
drwxrwxr-x+ 5 tomcat     phet 133 Nov 12  2016 1.1.2-phetiophetio/ => @samreid 
drwxrwxr-x+ 5 tomcat     phet 134 Oct  5  2016 1.1.1-phetio/ => @zepumph 
drwxrwxr-x+ 6 tomcat     phet 151 Sep 30  2016 1.1.0-phetio/ => @zepumph 
drwxrwsr-x+ 5 tomcat     phet 207 Sep 28  2016 1.1.0-phetio.4/ => @mattpen
drwxrwxr-x+ 4 tomcat     phet 118 Sep 24  2016 1.1.0-phetiodev.3/ => @samreid 
drwxrwsr-x+ 3 tomcat     phet 137 Aug 29  2016 1.1.0-dev.2/ => @samreid 
drwxrwsr-x+ 3 tomcat     phet 137 Aug 23  2016 1.1.0-dev.1/ => @samreid 
```

Note that the syntax for these version numbers is all over the place.

Two of the 1.1 dev versions have no associated commits in git history. And I see git history for these 1.1 versions that do not appear on bayes:

1.1.3-rc.2 => @zepumph 6/28/18
1.1.3-phetiorc.2 => @zepumph 6/28/18
1.1.3-rc.4 => @jonathanolson 6/29/18
1.1.3-rc.5 => @jonathanolson 7/6/18

Do any of you know what was going on here?

**Comment by pixelzoom** [comment]
Perhaps this was related https://github.com/phetsims/function-builder/issues/119, a problem with the Record Wrapper. Maybe that doesn't require PhET-iO instrumentation?

**Comment by pixelzoom** [comment]
The published version is 1.0.x, so I was expecting to be publishing 1.1.x.  But when I went to publish the dev version of Function Builder, it was 1.2.x.   That's confusing and concerning.  We'll investigate in https://github.com/phetsims/function-builder/issues/140.

In the meantime, I'll advance both function-builder and function-builder-basics to 1.2, and proceed with dev deploy.

**Comment by pixelzoom** [comment]
https://phet-dev.colorado.edu/html/function-builder/1.2.0-dev.5/phet/function-builder_en_phet.html

**Comment by pixelzoom** [comment]
Higher-priority issues were added to the QA queue, so https://github.com/phetsims/qa/issues/766 and https://github.com/phetsims/qa/issues/767 are on hold, and will require new dev versions.  QA will contact me when they are ready to proceed.

**Comment by pixelzoom** [comment]
@KatieWoe said in Slack:
> There should be room for the function builder devs now

I published new dev versions:
* [1.2.0-dev.6](https://phet-dev.colorado.edu/html/function-builder/1.2.0-dev.6/phet/function-builder_all_phet.html) for https://github.com/phetsims/qa/issues/766
* [1.2.0-dev.2](https://phet-dev.colorado.edu/html/function-builder-basics/1.2.0-dev.2/phet/function-builder-basics_all_phet.html) for https://github.com/phetsims/qa/issues/767

**Comment by pixelzoom** [comment]
Slack to @amanda-phet and @kathy-phet:

Chris Malley 4:00 PM
QA has completed dev testing for Function Builder & Function Builder: Basics.  There are 3 issues, none sim-specific:
https://github.com/phetsims/function-builder/issues/142
https://github.com/phetsims/function-builder/issues/143
https://github.com/phetsims/function-builder/issues/144
Kathy decided to wait for a fix for https://github.com/phetsims/function-builder/issues/142.  And there have been a very large number of common-code changes (TypeScript conversions) since I published these dev versions.  So I’m concerned that creating a release branch (even today, much less next week) is likely to pick up common-code regressions that will need to be patched during RC testing.
How would you like to proceed? (edited) 

Kathy Perkins 4:37 PM
Let's see when 142 is fixed, and then we can do a dev test. Not bad idea to test for regressions with another dev.

**Comment by pixelzoom** [comment]
Status as of 2/16/2022 @ 4:17PM MST:

* https://github.com/phetsims/phet-core/issues/103 block publication, in progress by @jonathanolson 
* https://github.com/phetsims/function-builder/issues/142 blocks sim publication, fixed by @jessegreenberg, pending test by @Nancy-Salpepi 
* https://github.com/phetsims/function-builder/issues/143 block sim publication, assigned to @kathy-phet 
* https://github.com/phetsims/function-builder/issues/144 blocks sim publication, assigned to @amanda-phet

**Comment by pixelzoom** [comment]
2/17/22 design meeting:

TypeScript conversion is the priority for Q1 2022, and it's therefore difficult to minimize common-code changes between the time of publishing dev version for testing, and creating release branches after testing has completed.  So we decided to pause further work on this issue until the beginning of Q2.  Around 4/1, PhET is planning to re-deploy a batch of PhET-iO sims, and there will be a window around that time during which common-code changes will be paused.  At that time, we'll begin a new (full) dev test for this sim.

**Comment by pixelzoom** [comment]
On hold until PHET-iO batch release https://github.com/orgs/phetsims/projects/44 happens.

**Comment by pixelzoom** [comment]
At 5/12/22 dev meeting, @kathy-phet asked me to decouple this from the PhET-iO batch release, which is delayed.  

On 5/19/22, @KatieWoe said that QA has capacity to do dev testing.  I published these dev tests:

* https://github.com/phetsims/qa/issues/804
* https://github.com/phetsims/qa/issues/805

**Comment by pixelzoom** [comment]
Slack#design-sims 5/26/2022:

> [Chris Malley](https://app.slack.com/team/U6EMFARV2)  [8:53 AM](https://phetsims.slack.com/archives/C5PSTV77B/p1653576839640759)
Since I’m out today, here’s a quick “check in” update on Function Builder.
Dev testing was completed for 1.2 release.  Most of the issues were common-code issues, and all but 1 have been addressed. https://github.com/phetsims/function-builder/issues/148 is a showstopper: the Screenshot feature is crashing on iPad Air 2.  This issue will require [@jonathanolson](https://phetsims.slack.com/team/U6E4HHSTF) to investigate, and it’s unclear when he’ll get a test device, or how this will be prioritized.  This problem is almost certainly not specific to FB, so a batch MR may be required.
So… I’m not sure about how to proceed with FB.  It’s been 1 week since I created the dev tests, and there have been many changes to master. If we further delay creating the 1.2 branch, then we’ll need to repeat dev testing after #148 is fixed.  If I create 1.2 now, we can continue with RC testing, but will need to apply a patch for #148 (or possibly publish with the crash, and fix via a batch MR).
[@kathy](https://phetsims.slack.com/team/U98Q0AX7B) [@amanda](https://phetsims.slack.com/team/U5PSTB1L1) thoughts?
> 
> [Kathy Perkins](https://app.slack.com/team/U98Q0AX7B)  [11:10 AM](https://phetsims.slack.com/archives/C5PSTV77B/p1653585006638289)
Please continue and create the rc. Thanks

RC tests published:
* https://github.com/phetsims/qa/issues/807
* https://github.com/phetsims/qa/issues/808

**Comment by pixelzoom** [comment]
We decidd to do another RC spot-check, to patch in #150 (Screenshot crash).  

RC tests published 6/10/2022:
* https://github.com/phetsims/qa/issues/813
* https://github.com/phetsims/qa/issues/813

**Comment by pixelzoom** [comment]
In order to update QA credits for #141, I patched function-builder 1.2 and function-builder-basics 1.2, then published 1.2.0-rc.3 for both sims.  As with last-minute credits updates that have been done in the past, these RCs were not QA tested.  I spot-checked the About dialog, to make sure the credits looked correct.
